astronomer/providers/snowflake/hooks/sql_api_generate_jwt.py

```diff
@@ -0,0 +1,168 @@
+import base64
+import hashlib
+import logging
+from datetime import datetime, timedelta, timezone
+
+# This class relies on the PyJWT module (https://pypi.org/project/PyJWT/).
+import jwt
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+
+logger = logging.getLogger(__name__)
+
+try:
+    from typing import Any, Optional, Text
+except ImportError:  # pragma: no cover
+    logger.debug(
+        "# Python 3.5.0 and 3.5.1 have incompatible typing modules.", exc_info=True
+    )  # pragma: no cover
+    from typing_extensions import Text  # pragma: no cover
+
+ISSUER = "iss"
+EXPIRE_TIME = "exp"
+ISSUE_TIME = "iat"
+SUBJECT = "sub"
+
+# If you generated an encrypted private key, implement this method to return
+# the passphrase for decrypting your private key. As an example, this function
+# prompts the user for the passphrase.
+
+
+class JWTGenerator(object):
+    """
+    Creates and signs a JWT with the specified private key file, username, and account identifier.
+    The JWTGenerator keeps the generated token and only regenerates the token if a specified period of time has passed.
+    """
+
+    LIFETIME = timedelta(minutes=59)  # The tokens will have a 59 minute lifetime
+    RENEWAL_DELTA = timedelta(minutes=54)  # Tokens will be renewed after 54 minutes
+    ALGORITHM = "RS256"  # Tokens will be generated using RSA with SHA256
+
+    def __init__(
+        self,
+        account: Text,
+        user: Text,
+        private_key: Any,
+        lifetime: timedelta = LIFETIME,
+        renewal_delay: timedelta = RENEWAL_DELTA,
+    ):
+        """
+        Creates an object that generates JWTs for the specified user, account identifier, and private key
+
+        :param account: Your Snowflake account identifier.
+            See https://docs.snowflake.com/en/user-guide/admin-account-identifier.html. Note that if you are using
+            the account locator, exclude any region information from the account locator.
+        :param user: The Snowflake username.
+        :param private_key: Private key from the file path for signing the JWTs.
+        :param lifetime: The number of minutes (as a timedelta) during which the key will be valid.
+        :param renewal_delay: The number of minutes (as a timedelta) from now after which the JWT
+            generator should renew the JWT.
+        """
```

Comment:
Shouldn't it params comes near class rather than init?

Reply:
Doesn't make any difference for me it can be in init as well